boards.txt

```diff
@@ -5716,6 +5716,362 @@ hornbill32minima.menu.DebugLevel.debug.build.code_debug=4
+dfrobot_beetle_esp32c3.menu.DebugLevel.error=Error
+dfrobot_beetle_esp32c3.menu.DebugLevel.error.build.code_debug=1
+dfrobot_beetle_esp32c3.menu.DebugLevel.warn=Warn
+dfrobot_beetle_esp32c3.menu.DebugLevel.warn.build.code_debug=2
+dfrobot_beetle_esp32c3.menu.DebugLevel.info=Info
+dfrobot_beetle_esp32c3.menu.DebugLevel.info.build.code_debug=3
+dfrobot_beetle_esp32c3.menu.DebugLevel.debug=Debug
+dfrobot_beetle_esp32c3.menu.DebugLevel.debug.build.code_debug=4
+dfrobot_beetle_esp32c3.menu.DebugLevel.verbose=Verbose
+dfrobot_beetle_esp32c3.menu.DebugLevel.verbose.build.code_debug=5
+
+##############################################################
+
+dfrobot_firebeetle2_esp32s3.name=DFRobot Firebeetle 2 ESP32-S3
+#dfrobot_firebeetle2_esp32s3.vid.0=0x3343
+#dfrobot_firebeetle2_esp32s3.pid.0=0x83CF
+dfrobot_beetle_esp32c3.vid.0=0x303a
+dfrobot_beetle_esp32c3.pid.0=0x1001
+
+dfrobot_firebeetle2_esp32s3.bootloader.tool=esptool_py
+dfrobot_firebeetle2_esp32s3.bootloader.tool.default=esptool_py
+
+dfrobot_firebeetle2_esp32s3.upload.tool=esptool_py
+dfrobot_firebeetle2_esp32s3.upload.tool.default=esptool_py
+dfrobot_firebeetle2_esp32s3.upload.tool.network=esp_ota
+
+dfrobot_firebeetle2_esp32s3.upload.maximum_size=1310720
+dfrobot_firebeetle2_esp32s3.upload.maximum_data_size=327680
+dfrobot_firebeetle2_esp32s3.upload.flags=
+dfrobot_firebeetle2_esp32s3.upload.extra_flags=
+dfrobot_firebeetle2_esp32s3.upload.use_1200bps_touch=false
+dfrobot_firebeetle2_esp32s3.upload.wait_for_upload_port=false
+
+dfrobot_firebeetle2_esp32s3.serial.disableDTR=false
+dfrobot_firebeetle2_esp32s3.serial.disableRTS=false
+
+dfrobot_firebeetle2_esp32s3.build.tarch=xtensa
+dfrobot_firebeetle2_esp32s3.build.bootloader_addr=0x0
+dfrobot_firebeetle2_esp32s3.build.target=esp32s3
+dfrobot_firebeetle2_esp32s3.build.mcu=esp32s3
+dfrobot_firebeetle2_esp32s3.build.core=esp32
+dfrobot_firebeetle2_esp32s3.build.variant=dfrobot_firebeetle2_esp32s3
+dfrobot_firebeetle2_esp32s3.build.board=ESP32S3_DEV
+#dfrobot_firebeetle2_esp32s3.build.board=DFROBOT_FIREBEETLE_2_ESP32S3
+
+dfrobot_firebeetle2_esp32s3.build.usb_mode=1
+dfrobot_firebeetle2_esp32s3.build.cdc_on_boot=0
+dfrobot_firebeetle2_esp32s3.build.msc_on_boot=0
+dfrobot_firebeetle2_esp32s3.build.dfu_on_boot=0
+dfrobot_firebeetle2_esp32s3.build.f_cpu=240000000L
+dfrobot_firebeetle2_esp32s3.build.flash_size=4MB
+dfrobot_firebeetle2_esp32s3.build.flash_freq=80m
+dfrobot_firebeetle2_esp32s3.build.flash_mode=dio
+dfrobot_firebeetle2_esp32s3.build.boot=qio
+dfrobot_firebeetle2_esp32s3.build.boot_freq=80m
+dfrobot_firebeetle2_esp32s3.build.partitions=default
+dfrobot_firebeetle2_esp32s3.build.defines=
+dfrobot_firebeetle2_esp32s3.build.loop_core=
+dfrobot_firebeetle2_esp32s3.build.event_core=
+dfrobot_firebeetle2_esp32s3.build.flash_type=qspi
+dfrobot_firebeetle2_esp32s3.build.psram_type=qspi
+dfrobot_firebeetle2_esp32s3.build.memory_type={build.flash_type}_{build.psram_type}
+
+dfrobot_firebeetle2_esp32s3.menu.PSRAM.disabled=Disabled
```

Comment:
do you really need all available flash/psram options? what modules are on the board?